docs/pages/api/access-plugin.mdx

```diff
@@ -0,0 +1,738 @@
+|---|---|
+|`github.com/gravitational/teleport-plugins/lib`|Code shared between Teleport plugins, e.g., to run a background job for retrieving new events from the Teleport Auth Service.|
+|`github.com/gravitational/teleport-plugins/lib/watcherjob`|A library for writing programs that listen for audit events from the Auth Service.|
+|`github.com/gravitational/teleport/api/client`|A library for authenticating to the Auth Service's gRPC API and making requests.|
+|`github.com/gravitational/teleport/api/types`|Types used in the Auth Service API, e.g., Access Requests.|
+|`github.com/gravitational/trace`|Presenting errors with more useful detail than the standard library provides.|
+|`google.golang.org/api/option`|Settings for configuring Google API clients.|
+|`google.golang.org/api/sheets/v4`|The Google Sheets API client library, aliased as `sheets` in our program.|
+|`google.golang.org/grpc`|The gRPC client and server library.|
+
+### Make some initial declarations
+
+After the `import` block, add the following declarations:
+
+```go
+const (
+	proxyAddr     string = ""
+	initTimeout          = time.Duration(30) * time.Second
+	spreadSheetID string = ""
+)
+
+var requestStates = map[types.RequestState]string{
+	types.RequestState_APPROVED: "APPROVED",
+	types.RequestState_DENIED:   "DENIED",
+	types.RequestState_PENDING:  "PENDING",
+	types.RequestState_NONE:     "NONE",
+}
+
+type googleSheetsPlugin struct {
+	sheetsClient   *sheets.SpreadsheetsService
+	teleportClient *client.Client
+}
+```
+
+`proxyAddr` indicates the address and port of your Teleport Proxy Service or
+Teleport Enterprise Cloud tenant. `initTimeout` is the maximum time we will wait
+to connect to the Auth Service, in this case, 30 seconds.
+
+Assign `spreadSheetID` to the ID of the spreadsheet you created earlier. To find
+the spreadsheet ID, visit your spreadsheet in Google Drive. The ID will be in
+the URL path segment called `SPREADSHEET_ID` below:
+
+```text
+https://docs.google.com/spreadsheets/d/SPREADHSEET_ID/edit#gid=0
+```
+
+Access Requests have one of four states: approved, denied, pending, and none.
+Later in the guide, we will need a way to map these states to strings that we
+can write to our Google Sheets spreadsheet. We obtain the request states from
+Teleport's `types` library and map them to strings in the `requestStates` map.
+
+`googleSheetsPlugin` is a struct type that contains the API clients for the
+Teleport Auth Service and Google Sheets. We will explain later why we use a
+struct to organize these two clients, instead of, say, two separate variables.
+
+### Prepare row data
+
+Whether creating a new row of the spreadsheet or updating an existing one, we
+need a way to extract data from an Access Reqeust in order to provide it to
+Google Sheets. Below the `googleSheetsPlugin` type declaration, add the
+following function:
+
+```go
+func (g *googleSheetsPlugin) makeRowData(ar types.AccessRequest) *sheets.RowData {
+
+	t := ar.GetCreationTime().String()
```

Comment:
I find this hard to follow, since the variables have non-descriptive names and are used a dozen or so lines after they are declared.

How about a helper:

```
func stringPtr(s string) *string { return &s 
```

Then you don't need these variables here and you inline them below.

```
UserEnteredValue: &sheets.ExtendedValue{
    StringValue: stringPtr(ar.GetCreationTime().String()),
},
```

Reply:
Added for the one-line assignments. For the request state and link string, I used descriptive variable names instead.